src/operator/cudnn_rnn-inl.h

```diff
@@ -498,6 +498,16 @@ class CuDNNRNNOp : public Operator{
         }
         CUDNN_CALL(cudnnSetRNNMatrixMathType(rnn_desc_, math_type));
       #endif
+      if (param_.projection_size.has_value()) {
+      #if (CUDNN_MAJOR == 7 && CUDNN_MINOR >= 1) || CUDNN_MAJOR > 7
+        CUDNN_CALL(cudnnSetRNNProjectionLayers(s->dnn_handle_,
+                                               rnn_desc_,
+                                               param_.projection_size.value(),
+                                               0));
+      #else
+        LOG(FATAL) << "Projection is only available with later than CuDNN 7.1.1";
```

Comment:
does the projection work for all RNN types? if so, add the support to all RNN layers. if not, could you check for the type here and make sure it's supported?

Reply:
Okay